src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ValueForms/FirstLowerCaseValueFormModel.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.ValueForms
+{
+    public class FirstLowerCaseValueFormModel : IValueForm
+    {
+        public string Identifier => "firstLowerCase";
+
+        public string Name { get; }
+
+        public FirstLowerCaseValueFormModel()
+        {
+        }
+
+        public FirstLowerCaseValueFormModel(string name)
+        {
+            Name = name;
+        }
+
+        public IValueForm FromJObject(string name, JObject configuration)
+        {
+            return new FirstLowerCaseValueFormModel(name);
+        }
+
+        public string Process(IReadOnlyDictionary<string, IValueForm> forms, string value)
+        {
+            switch (value)
+            {
+                case null: return null;
+                case "": return value;
+                default: return value.First().ToString().ToLower() + value.Substring(1);
```

Comment:
`ToLower` is culture-dependent. Is this what we want or should we change it to use `CultureInfo.InvariantCulture`?

Reply:
Same goes for `ToUpper` below.

Reply:
There was a question whether we need invariant versions above. We do have lowerCase and lowerCaseInvariant forms.

To be honest I'm not quite sure if we should do two versions here. I believe that this form is needed in invariant culture, as I see the use case to generate variables names, which are usually invariant.  However, keeping firstLowerCase as Invariant and lowerCase as culture specific might be confusing. 

Given said that, I would rather do 2 forms (firstLowerCase and firstLowerCaseInvariant) and pass the responsibility to template author which one to choose (and if it is used correctly in given context :) ).

Reply:
Implemented invariant forms in new commit and updated test.